lib/ansible/module_utils/network/exos/config/vlans/vlans.py

```diff
@@ -0,0 +1,271 @@
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        requests = []
+
+        request_patch = deepcopy(self.VLAN_PATCH)
+        request_body = deepcopy(self.REQUEST_BODY)
+        have_copy = []
+        for w in want:
+            if w.get('vlan_id'):
+                h = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+                if h:
+                    if dict_diff(w, h):
+                        request_body = deepcopy(self.REQUEST_BODY)
+                        request_body = self._update_vlan_config_body(w, request_body)
+                        request_patch["data"]["openconfig-vlan:vlans"]["vlan"].append(request_body)
+                    have_copy.append(h)
+                else:
+                    request_post = deepcopy(self.VLAN_POST)
+                    request_body = self._update_vlan_config_body(w, request_body)
+                    request_post["data"]["openconfig-vlan:vlans"].append(request_body)
+                    request_post["data"] = json.dumps(request_post["data"])
+                    requests.append(request_post)
+
+        for h in have:
+            request_delete = deepcopy(self.VLAN_DELETE)
+            if h not in have_copy and h['vlan_id'] != 1:
+                request_delete["path"] = self.DEL_PATH + str(h['vlan_id'])
+                requests.append(request_delete)
+
+        if len(request_patch["data"]["openconfig-vlan:vlans"]["vlan"]):
+            request_patch["data"] = json.dumps(request_patch["data"])
+            requests.append(request_patch)
+
+        return requests
+
+    def _state_merged(self, want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        requests = []
+
+        request_body = deepcopy(self.REQUEST_BODY)
+
+        for w in want:
+            if w.get('vlan_id'):
+                h = search_obj_in_list(w['vlan_id'], have, 'vlan_id')
+                if h:
+                    continue
+                else:
+                    request_post = deepcopy(self.VLAN_POST)
+                    request_body = self._update_vlan_config_body(w, request_body)
+                    request_post["data"]["openconfig-vlan:vlans"].append(request_body)
+                    request_post["data"] = json.dumps(request_post["data"])
+                    requests.append(request_post)
+
+        return requests
+
+    def _state_deleted(self, want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        requests = []
+
+        if want:
+            for w in want:
+                request_delete = deepcopy(self.VLAN_DELETE)
```

Comment:
Move this line after 250.
You won't need to deepcopy unless you are actually creating a request.